python/tk_multi_publish2/api/plugins/collector_instance.py

```diff
@@ -117,6 +117,28 @@ def run_create_properties_widget(self, parent, items):
                 extra=_get_error_extra_info(error_msg)
             )
 
+    def run_on_properties_changed(self, items):
+        """
+        Method that runs when the property_changed signal is fired from the property widget.
+
+        .. note:: This method is a no-op if running without a UI present
+
+        :param items:  List of items to run property change hook for.
+        """
+
+        # nothing to do if running without a UI
+        if not sgtk.platform.current_engine().has_ui:
```

Comment:
This is not technically a UI-specific method. A developer could potentially trigger this via the publisher api in non-UI mode, so we probably shouldn't have this check.